### DISCUSSION TITLE
Weird Issue with Instagram Vids

### DISCUSSION THREAD

**Discussion by artistwantab** [aclap-dev/video-downloadhelper#853] [Help and Q&A]
### Your question

I use Vid-DH to download vids I find relevant to my new job.  

Recently I am unable to download more then 4-5 vids before the program starts acting weird.  

The first few vids is no issue.   I hit the VDH tab and then click download and the video downloads completely.

After a few vids when I get the tab instead of one download option I get numerous for the same video and none actually work as a video.   Some files a 0KB and some are 3mb.

I have tried using a VPN thinking its instagram not allowing my ISP to download more then a few vids.   

Tried the following also....

logging in to Insta and out --- same results
Clearing Cache
created a blank profile to see if any other apps were causing an issue
tried numerous vids

Using Firefox Latest 
VDH Latest
Win 10

Thank you in advance any help is appreciated.

Here is an example of a vid

https://www.instagram.com/reel/C61-4eGumgL/?igsh=MTc4MmM1YmI2Ng%3D%3D

### Addon & CoApp version and details

version: 9.0.2.3
target: mozilla
channel: stable
lang: en-US
coapp: "checking"
license: {"unset":true}
platform: x86-32 win
UA: Mozilla/5.0 (Windows NT 10.0; rv:126.0) Gecko/20100101 Firefox/126.0
{
    "used_sidebar_button": true,
    "used_popup_button": true,
    "never_show_no_incognito_msg_again": true,
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "use_sidebar": false,
    "open_count_store": 11,
    "successfull_dl": 5
}

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.  Also, I am not a member of Instagram & never intend to become one.  In fact, I am not a member of any so-called "social" network.

Now here's what I did.  This is just something that works.  You can use it or not, your choice.  I tried to download this with VDH & I just got an error.  There were about a dozen or more entries on the VDH menu, all of them marked HTTPMedia, none of them showing a duration.  They had various sizes listed, between a high of 4M & a low of 0K.  I tried the first one listed & I just got an error from VDH.  The errors now don't actually contain any information.  Supposedly if you click the button to submit the error, it goes somewhere, I don't know where, & I don't know what happens to it.  That's a shame because it preempts the ability of folks here from at least trying to help.  But there it is.

The HTTPMedia marking means VDH is detecting only individual files.  It did not find a streaming manifest that would have allowed VDH to just download the stream.  The web site is supplying only individual chunks of the stream without the information VDH needs to assemble the chunks into a complete video.  This is not the fault of VDH.  This is the way the web site is serving the information to the client browser.

By looking at the Network Monitor, I discovered that there were several MP4 items sent from Instagram to my browser (Firefox).  You can double-click any entry in the Network Monitor & Firefox will attempt to open that object in its own tab.  Here's what I was looking at:

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/c6ebe80f-65a7-40df-92b6-2cfa0ed2a6b4)

In that image, I have the fourth MP4 entry highlighted.  I'll explain why in a moment.  For now, focus on the first MP4 entry.  What I am describing is a trial & error process.  I just started with the first one & was prepared to simply continue with each one until I found something useful.  Today is, apparently, my lucky day because the first one gave me something useful.  A few moments before I captured that screenshot, I double-clicked that first entry.  That opened that MP4 in its own browser tab.  That just generated an error in Firefox.  But I captured the URL of that first entry from the URL bar of the browser window.  That URL was this:

```
https://scontent-ord5-2.cdninstagram.com/o1/v/t16/f1/m82/DF4D75464953989E2B300F3C2B857E8B_video_dashinit.mp4?efg=eyJ2aWRlb19pZCI6bnVsbCwidmVuY29kZV90YWciOiJpZy14cHZkcy5jbGlwcy5jMi1DMy5kYXNoX2Jhc2VsaW5lXzFfdjEifQ&_nc_ht=scontent-ord5-2.cdninstagram.com&_nc_cat=102&ccb=9-4&oh=00_AYDgLzFFOY_wk-pF9NUco-RPBho_dRHnbkMgW39wO4qouQ&oe=66682E4C&_nc_sid=9ca052&bytestart=867&byteend=1042
```

That is an ugly monstrosity, typical of the ugly monstrosities that most web sites generate.  But this ugly monstrosity actually has something interesting to look at.  It ends with this: `&bytestart=867&byteend=1042`.   So I removed that bit to give me this still ugly but more useful monstrosity:

```
https://scontent-ord5-2.cdninstagram.com/o1/v/t16/f1/m82/DF4D75464953989E2B300F3C2B857E8B_video_dashinit.mp4?efg=eyJ2aWRlb19pZCI6bnVsbCwidmVuY29kZV90YWciOiJpZy14cHZkcy5jbGlwcy5jMi1DMy5kYXNoX2Jhc2VsaW5lXzFfdjEifQ&_nc_ht=scontent-ord5-2.cdninstagram.com&_nc_cat=102&ccb=9-4&oh=00_AYDgLzFFOY_wk-pF9NUco-RPBho_dRHnbkMgW39wO4qouQ&oe=66682E4C&_nc_sid=9ca052
```

By editing the URL in the URL bar of that browser tab, then hitting enter, a video showed up:

![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/659ccfb8-e971-4136-b965-fb95a369e6a1)

The little speaker with a slash through it indicates that there is no audio with this video.  However, VDH showed a new entry on its menu & it downloaded in like nothing flat:

![#03](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/f881af0a-b4ae-4c25-85e1-2932ec8a3aae)

As expected, there is a video here but it has no audio track.

So now the search began.  I double-clicked each MP4 entry in the Network Monitor in turn.  I knew there had to be an audio track in there somewhere.  I just didn't know where.  The only way to find it was to double-click each MP4 entry in turn until I found it.  The second & third MP4s were the same video again.  Double-clicking the fourth entry, the one highlighted in the image above, got me this in a new browser tab, once again after editing the URL to remove the 2 pieces from its end:

![#04](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/4f77d8ca-2cac-47eb-b909-016d8241b47f)

It's pretty obvious that this is an audio track with no video track.  VDH once again recognized something new & it also downloaded no problem:

![#05](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/9064dee9-5518-449c-96ef-ed7d72123895)

Once again, as expected.  No video track, but an audio track.

I played these 2 files synchronously in VLC using its synchronous playback feature.  No problems.  Video with audio right to the end.

I want to emphasize that this is a trial & error process.  I was quite prepared to double-click every MP4 in the Network Monitor.  You just have to be persistent.  In other cases, you might not find your tracks in the first & fourth MP4s.  But they will be there.  I have to assume that.  Of course, I'm projecting that hypothesis based on just this one data point.  That isn't very scientific.  But I have to assume Instagram does not custom tailor each page in a different format.  You will need to try this on different pages to see whether it will work.

But.  You will, of course, raise a but.  But you downloaded a bunch of videos successfully before you encountered this error.  Hmmm.......  Maybe my assumption is wrong that Instagram does not tailor each page in a different format.  It seems that they are that crazy.  Different pages appear to be constructed differently.  Maybe there was an old format that has been replaced by a new format.  But lots of pages in the old format are still around.  I don't know.  Like I say, I'm not a member of Instagram.  Personally, I think the world would be a much better place without these "social" network sites.  I think there's nothing at all social about them.  There's my rant.  It seems there are many formats of pages on Instagram.  Be thankful when VDH works on any given page.  But be prepared to resort to thinking outside the box on other pages.

**Comment by Mattmjs** [comment]
Or try a simpler solution  -->   **Ctrl +Shift + M**  to open emulation mode in the browser.
Then go to the video and it detects the complete file. I see it as a 22Mb file.

![10-06-24](https://github.com/aclap-dev/video-downloadhelper/assets/160297904/b8cf4452-9ad6-4686-ba42-a2b9e143e115)

**Comment by artistwantab** [comment]
@GrampaWildWilly 

Thank you.  I tried all of that hoping to locate the video in full

@Mattmjs  You got it to do what I can get it to do for about three video before it cuts of and starts acting up.

I tried emulation mode and sadly I got the same results.

**Reply by GrampaWildWilly** [reply]
Well, the full video is there, it's just dressed for a masked ball.  You have to unmask it by editing the URL.  Actually, 2 URLs, one for the video, one for the audio.  You will not find something that has both the video & the audio here.  They are separate.  That's how the web site presents its data.

Emulation mode.  I keep forgetting about that.  I tried it just now on the Stout video & it worked for me, too.  Windows 11 Pro 64-bit version 23H2 build 22631.3672, Firefox 126.0.1 64-bit, licensed VDH 9.0.2.3a1 beta, CoApp 2.0.19.

I notice in the original post these lines:

> coapp: "checking"
license: {"unset":true}

This says the CoApp is not installed properly.  If you have a license, it is recognized only if you have the CoApp functioning properly.  Since this is not HLS or DASH, I'm not sure having the CoApp installed properly makes a difference.  I have it installed & I suppose my download used it.  I don't know how this would affect you.

I don't know?  That's a command to figure it out.  So here's what I tried.  I went into the VDH settings & checked the box that said Do not use CoApp when possible.  That's the closest I'm willing to get to actually uninstalling my CoApp.  With emulation mode on, I visited your Instagram video page again.  VDH presented me with 14 items.  I downloaded them all.  Every one of them did NOT use the CoApp.  I had a bunch of entries in my Firefox download manager for these downloads.  That's what happens when VDH uses the browser instead of the CoApp to download things.  But the Windows properties for every one of these 14 files were empty.  No video information.  No audio information.  Yet VDH looked like it had completed the downloads successfully & it thought everything was fine.  I tried to play all 14 of these MP4s in VLC & not a single frame of video appeared in VLC.  It rejected them all out of hand.

Conclusion?

**Reply by GrampaWildWilly** [reply]
Forgot to mention.  Since I did some Reload addon functions mixed in amongst all my page reloads, it was necessary for me to actually launch playback of the Instagram video for a second to get VDH to recognize it.  That's not overly surprising.  So if you find your VDH menu goes empty at some point, try that.

**Reply by GrampaWildWilly** [reply]
Wait.  My initial test was invalid.  I carefully did this.

1. Visited the Instagram page.
2. Turned on emulation mode.
3. Turned on VDH setting for Don't use CoApp.
4. Reloaded addon.
5. Reloaded web page.  VDH menu remained empty.
6. Clicked play on the video & paused it after a second or two.
7. VDH recognized 1 item.
8. Downloaded it.  VDH acted like it worked.
9. Verified that the browser did the download & not the CoApp.
10. The file properties were empty.
11. VLC refused to play it.

After that, I did this:

1. Go into the VDH settings & turned off the Don't use CoApp setting.
2. Reload addon.
3. Still with emulation mode on, reload Instragram page.
4. VDH menu remained empty.
5. Clicked play then paused it after a second.
6. VDH recognized a single MP4 of size 22M.
7. Told VDH to download that.
8. Succeeded.
9. VLC played it perfectly.  Video + audio.

There.  Now it's a proper test.  Now I can challenge you properly.

Conclusion?

**Reply by GrampaWildWilly** [reply]
If you find a video that "acts up" as you put it, do this as a test.

1. Close all browser windows.
2. Come back to the same Instagram page not in emulation mode & try to download the video with VDH.

Does it still fail?

Next test.  Do the same thing but turn on emulation mode.

I'm trying to isolate whether it's the fact that you got some to work & then one failed or if it's just that video that happens to be a problem.  I want you to retry one that fails but do it in a session in which you haven't downloaded anything previously.  I don't think Reload addon is enough in this case.  I think you need to close all browser windows.

**Reply by GrampaWildWilly** [reply]
Maybe we need you to give us the series of URLs for which you get a few to work & then one fails.

**Comment by Mattmjs** [comment]
I think Instagram behaves like this, the first 3 or so videos they serve as a single mp4 files. Then after that just to be a pain
they decide we won't serve any additional videos this way but as smaller segments. This may be down to cookies, your ip address
or your Instagram account. I'm not a member of this site but those factors probably play a part in that.

**Reply by GrampaWildWilly** [reply]
In that case, as non-members, I guess we get treated to the "just to be a pain" situation.  My initial visit to that web page gave me the video in segments & I had to remove the segment identifers to get the URL of the full video track, & do the same thing to get the URL of the full audio track.  Don't you feel treated special @artistwantab that you actually get 3 easy videos before you hit a problem?

In light of what Matt says, my idea of closing all browser windows & retrying might actually be relevant.  You might be able to get away with just logging off & back on.  But it's possible you might have to BOTH log off AND close all browser windows.  Only you can figure it out, being a member there.
